### ISSUE TITLE
[Feature Request]: PLAF109 / Polar Wet Feeder | Missing Implementation for Feeding

### ISSUE THREAD

**Issue by goatrocks** [jjjonesjr33/petlibro#5]
### Device model name

Polar Wet Feeder

### Device model number

PLAF109

### Device type

Feeder

### What state, statistic or control do you want to be reported in the integration ?

most wanted:
meal schedules
if it is currently feeding

also nice:
to be able to trigger the feeder to feed now
rotate bowl
see when cat has eaten

I have two of these feeders and am happy to try to get set up to capture API calls :)

### Available to help integrate the new device

- [X] I agree to help with the integration of the new device by providing the necessary information that will be requested

**Comment by jjjonesjr33** [comment]
Hey @goatrocks would you be able to provide information on what the api pulls or give temporary `shared` access to this device via my support account? If you okay with that, I have a test account setup via `support@jonesinc.dev` that you can give access to via the Petlibro app on your phone.

Please let me know when you/if you do as there is a 30 min window for me to accept the shared device.

**Comment by jjjonesjr33** [comment]
Thanks @goatrocks I have it added now and will take a look at it later today and see what I can get added in for you!

**Comment by jjjonesjr33** [comment]
Just to give you an update @goatrocks I have the device loaded and some stats are pulling. There seems to be an issue with the feeding plan to disable/enable. I think this is due to the device being setup in a 3 meal/day cycle. When disabling a day after it has already processed the meal for that day it goes into an unknown state. Vs what I would rather have it do is disable the next meal. or all the meals like how it does for the other feeders.

There are a few other sensors I want to get added/adjusted. Once I get those done I will let you know. Thanks again for sharing the device so I can pull the data! 😄

**Comment by Sausagerolls** [comment]
Hi there, you say it pulls data in from the polar, but I've just set up a polar and there's nothing appearing in home assistant. Am I missing something or is it not integrated in yet?

**Comment by goatrocks** [comment]
Thank you so much for adding this!

**Comment by dennismdejong** [comment]
**Proposal: PLAF109 feeding — status & remaining gaps**

Status update on this issue (from Oct 2024):

**Already implemented (polar_wet_food_feeder.py):**
- Feed now, rotate food bowl, ring bell buttons (PR #50, v1.0.17)
- `feeding_plan_state`, `next_feeding_day/time/end_time`
- `manual_feed_id`, `manual_feed_now`
- `plate_position`, `temperature`, lid status
- PR #249 (merged) improved `feeding_plan_state` via `wetFeedingPlan` data

**Still open:**
- Confirming "is currently feeding" in real time
- Seeing when a pet has actually eaten (per-pet eating detection)
- Full schedule CRUD (create/edit schedules from HA)

**Proposed next steps:**
1. Ask the author for a `realInfo` capture while feeding (motor/runningState) so we can build a reliable "currently feeding" sensor.
2. Per-pet eating detection once the per-pet data path from #229 also covers the Polar, or via `workRecord` events.
3. Schedule CRUD is the larger piece — could be its own track.

@jjjonesjr33 — is the base functionality (feed + schedule status) now sufficient to largely close #5, with "currently feeding" + per-pet eating as the remaining items? And shall I post a capture checklist for the author?

**Comment by jjjonesjr33** [comment]
@dennismdejong Yeah, I think we are at a good point where we can consider the original scope of #5 covered and close this one out.

The main things I would still like to get figured out are the real-time **currently feeding** state and the **per-pet eating detection**. I agree that the full schedule CRUD would probably be better handled separately since that is going to be a much larger piece.

Go ahead and post the capture checklist for the author. Getting a `realInfo` capture while the feeder is actively running would be a good next step so we can see if there is a reliable motor/running state we can use.

I'm going to close this issue out, and we can track the remaining items separately as enhancements.

**Comment by goatrocks** [comment]
Just want to note that unfortunately I had to stop using this feeder because it does not keep food safely cooled, and it can make your cat sick. When I was using the integration it worked great, but the feeder is not something I would recommend people use at this point.
